site/content/contribute/server/style-guide.md

```diff
@@ -0,0 +1,98 @@
+
+This does not, however, mean that a developer has to fix any surrounding code to fix the style issues too. It’s encouraged to keep fixing things as we go, but it's not compulsory to do so. There's also no need for a reviewer to ask for that change if it's not something that's added/edited in that PR.
+
+The following goes into some of the specific anti-patterns that have crept in our codebase and we should aim to correct. The rules which are exempt from being applied in the model package are specifically indicated.
+
+## Guidelines
+
+### Default to sync instead of async
+
+Always prefer synchronous functions by default. Async calls are hard to get right. They have no control over goroutine lifetimes and introduce data races. If you think something needs to be asynchronous, measure it and prove it. Ask these questions:
+- Does it improve performance? If so, by how much?
+- What’s the tradeoff of the happy path vs. slow path?
+- How do I propagate errors?
+- What about back-pressure?
+- What should be my concurrency model?
+
+Avoid creating one-off goroutines without knowing when/how they exit. They cause problems that are hard to debug, and can often cause performance degradation rather than an improvement. Have a look at:
+- https://github.com/golang/go/wiki/CodeReviewComments#goroutine-lifetimes
+- https://github.com/golang/go/wiki/CodeReviewComments#synchronous-functions
+
+### Pointers to slices (exempt from model)
+
+Avoid using pointers to slices. Slices are already reference types which point to an underlying array. If you want a function to modify a slice, then return that slice from the function, rather than passing a pointer.
+
+### [CamelCase variables/constants](https://github.com/golang/go/wiki/CodeReviewComments#mixed-caps) (exempt from model)
+
+We prefer to use names like WebsocketEventPostEdited, not WEBSOCKET_EVENT_POST_EDITED.
+
+### Empty string check
+
+Use == "" to check empty strings, not len(s) == 0. This will be enforced with a linter soon.
+
+### [Reduce indentation](https://github.com/golang/go/wiki/CodeReviewComments#indent-error-flow)
+
+If there are multiple return statements in an if-else statement, remove the else block and outdent it.
+
+This is a sample code from `mlog/human/parser.go`:
+
+```go
+// Look for an initial "{"
+if token, err := dec.Token(); err != nil {
+	return result, err
+} else {
+	d, ok := token.(json.Delim)
+	if !ok || d != '{' {
+		return result, fmt.Errorf("input is not a JSON object, found: %v", token)
+	}
+}
+```
+
+This can be simplified to:
+
+```go
+// Look for an initial "{"
+if token, err := dec.Token(); err != nil {
+	return result, err
+}
+d, ok := token.(json.Delim)
+if !ok || d != '{' {
+	return result, fmt.Errorf("input is not a JSON object, found: %v", token)
+}
+```
+
+### Avoid creating more ToJSON methods
+
+We should avoid creating ToJSON methods for model structs. And just use `json.Marshal` at the call site. This has two main benefits:
```

Comment:
```suggestion
Do not create new `ToJSON()` methods for model structs. Instead, just use `json.Marshal` at the call site. This has two major benefits:
```

Reply:
The `()` is actually not used for method/function documentation. :P I learnt this the hard way in a code review. `ToJSON` is the actual name of the method.

Reply:
Ah OK, good to know. Please ignore those bits of my suggestions then :)